python/ray/worker.py

```diff
@@ -377,6 +377,14 @@ def retrieve_and_deserialize(self, object_ids, timeout, error_timeout=10):
                         timeout,
                         self.serialization_context)
                 return results
+            except pyarrow.lib.ArrowInvalid as e:
+                # TODO(ekl): the local scheduler could include relevant
+                # metadata in the task kill case for a better error message
+                invalid_error = RayTaskError(
+                    "<unknown>", None,
+                    "Invalid return value: likely worker died or was killed "
+                    "while executing the task.")
```

Comment:
We'll need to keep an eye out for this error and make sure that when we see it it actually corresponds to an actor death (as opposed to a bug in our serialization or something).